### PR TITLE
Frontend: Stop displaying the Pass Button

### PR DESCRIPTION
**Pull request by DanielNghiem**

### CODE REVIEW

File: hearts/static/room_client.js [hunk 1]
```diff
@@ -23,20 +23,16 @@ function RoomClient(socketClient, roomState) {
       } else {
         this.ui.renderPassButton(false);
       }
-    } else if (this.state.mode() === 'play') {
+    } else if (this.state.mode() === 'play'||'wait for play') {
```
Comment (j2kun) [inline hearts/static/room_client.js]: Should be 

```
this.state.mode() === 'play' || this.state.mode() ===  'wait for play'
```

File: hearts/static/room_socket_client.js [hunk 2]
```diff
@@ -15,15 +15,18 @@ function RoomSocketClient(apiUrl) {
     var data = {
       cards: cards
     };
-    var result = this.socket.emit('pass_cards', data);
-    return result.status;
+    this.socket.emit('pass_cards', data, function(result){
+      console.log('received confirmation', result);
```
Comment (j2kun) [inline hearts/static/room_socket_client.js]: If it's a failure, do you want to display the message? If so, this will be annoying, because you'll need to add a function argument to this method (`passCards`) that stores a callback for displaying the message. Then inside *this* callback you have to invoke *that* callback with the returned result.

This is the beginning of we call "callback hell"

File: hearts/static/room_ui.js [hunk 3]
```diff
@@ -67,7 +67,14 @@ function RoomUI(callbacks) {
       buttonHtml = ('<button id="pass_button">' +
         'Pass 3 cards ' + direction + '</button>');
       $('#status').html(buttonHtml);
-      $('#pass_button').click(this.passButtonClickCallback);
```
Comment (j2kun) [inline hearts/static/room_ui.js]: Isn't this one line equivalent to the "that = this" four lines below? What's the reason for the change?

File: hearts/static/room_ui.js [hunk 4]
```diff
@@ -67,7 +67,14 @@ function RoomUI(callbacks) {
...
+      var that = this;
+      $('#pass_button').click(function(event) {
+        that.passButtonClickCallback();
+      });
+    }
+    else {
```
Comment (j2kun) [inline hearts/static/room_ui.js]: style: `else` should be on the same line as preceding `}`

File: hearts/static/room_ui.js [hunk 5]
```diff
@@ -85,15 +92,24 @@ function RoomUI(callbacks) {
     $('#messages').html(messagesHtml);
   }
 
+  this.hideActionButtons = function(mode) {
+    // Hide the pass Button when in waiting mode
+    if (mode == 'wait for pass') {
+      $('#pass_button').hide();
+    }
+    // Should we make a Play Button and hide it too?
```
Comment (j2kun) [inline hearts/static/room_ui.js]: Playing a card will be simply clicking on the card picture.

File: hearts/static/room_ui.js (hunk 4 above)
Comment (j2kun) [reply hearts/static/room_ui.js]: This else statement should be unnecessary once you've added the `hideActionButtons` method, right?

If it is needed, I'd prefer having a single function that hides the pass button (`hideActionButtons` is probably fine for this), and calling it from here instead of duplicating the logic.

File: hearts/static/room_ui.js (hunk 3 above)
Comment (DanielNghiem) [reply hearts/static/room_ui.js]: Yes it is.  I changed it thinking that the callback for the click would do something else in addition to `passButtonClickCallback`.   I initially tried to hide the pass button here, but realized it didnt work here.  (Something about how javascript callbacks are asynchronous)

Comment (DanielNghiem) [reply hearts/static/room_ui.js]: I can change it back

File: hearts/static/room_socket_client.js (hunk 2 above)
Comment (DanielNghiem) [reply hearts/static/room_socket_client.js]: For now, I just want to log the message instead of displaying it.  
I will need to display messages when a card is played, though.  However, I don't want to use a callback to display the message either.

File: hearts/static/room_client.js (hunk 1 above)
Comment (DanielNghiem) [reply hearts/static/room_client.js]: Fixed.

File: hearts/static/room_ui.js (hunk 4 above)
Comment (DanielNghiem) [reply hearts/static/room_ui.js]: Fixed.

**Comment by j2kun** [comment]
LGTM
